test/extended/util/client.go

```diff
@@ -111,62 +109,63 @@ type resourceRef struct {
 	Name      string
 }
 
-// NewCLIWithFramework initializes the CLI using the provided Kube
-// framework. It can be called inside of a Ginkgo .It() function.
-func NewCLIWithFramework(kubeFramework *framework.Framework) *CLI {
-	cli := &CLI{
-		kubeFramework:           kubeFramework,
-		username:                "admin",
-		execPath:                "oc",
-		adminConfigPath:         KubeConfigPath(),
-		staticConfigManifestDir: StaticConfigManifestDir(),
+// CliOptions allows customization of the newely created CLI resource
+type CliOptions struct {
+	// KubeFramework to use, instead of creating a brand new one
+	KubeFramework *framework.Framework
+	// prefix for the created test namespace
+	BaseName string
+	// pod security level applied to the created e2e test namespace, defaults to `restricted`
+	PodSecurityEnforceLevel admissionapi.Level
+	// whether to add --namespace flag in oc invocations
+	WithoutNamespace bool
+	// whether we're dealing with HyperShift cluster
+	HyperShift bool
+}
+
+// NewCLI is responsible for creating a new CLI object for invoking OpenShift CLI,
+// CliOptions allow customization of the created object.
+func NewCLI(options CliOptions) *CLI {
```

Comment:
Perhaps you want `NewCLI(opts ...Option)`

```go
type Option func(cliOptions) cliOptions

func withTestFramework(framework) Option
func withPSALevel(level) Option
func withoutNamespace() Option

```

Reply:
Good idea, especially that we already have a lot of these `With*` methods.